docs/reference/search/rrf.asciidoc

```diff
@@ -1,6 +1,423 @@
+    },
+    "rank": {
+        "rrf": {
+            "window_size": 50,
+            "rank_constant": 20
+        }
+    }
+}
+----
+// TEST[skip:example fragment]
+
+In the above example, we first execute the knn search to get its global top 50 results.
+Then we execute the query to get its global top 50 results. Afterwards, on a coordinating
+node, we combine the the knn search results with the query results and rank them based on the
+rrf method to get the final top 10 results.
+
+Note that if `k` from a knn search is larger than `window_size`, the results are
+truncated to `window_size`. If `k` is smaller than `window_size`, the results are
+`k` size.
+
+[[rrf-supported-features]]
+==== Reciprocal rank fusion supported features
+
+RRF does support:
+
+* from
+* aggregations
+
+RRF does not currently support:
+
+* scrolling
+* point in time queries
+* sorting
+* rescoring
+* suggestions
+* highlighting
+* collapsable fields
+* explain
+* profile
+
+[[rrf-full-example]]
+==== Reciprocal rank fusion full example
+
+We begin by creating a mapping for an index with a text field, a vector field,
+and an integer field along with indexing several documents. For this example we
+are going to use a vector with only a single dimension to make the ranking easier
+to explain.
+
+[source,console]
+----
+PUT example-index
+{
+  "mappings": {
+        "properties": {
+            "text" : {
+                "type" : "text"
+            },
+            "vector": {
+                "type": "dense_vector",
+                "dims": 1,
+                "index": true,
+                "similarity": "l2_norm"
+            },
+            "integer" : {
+                "type" : "integer"
+            }
+        }
+    }
+}
+
+PUT example-index/_doc/1
+{
+    "text" : "rrf",
+    "vector" : [5],
+    "integer": 1
+}
+
+PUT example-index/_doc/2
+{
+    "text" : "rrf rrf",
+    "vector" : [4],
+    "integer": 2
+}
+
+PUT example-index/_doc/3
+{
+    "text" : "rrf rrf rrf",
+    "vector" : [3],
+    "integer": 1
+}
+
+PUT example-index/_doc/4
+{
+    "text" : "rrf rrf rrf rrf",
+    "integer": 2
+}
+
+PUT example-index/_doc/5
+{
+    "vector" : [0],
+    "integer": 1
+}
+
+POST example-index/_refresh
+----
+// TEST
+
+We now execute a search using RRF with a query, a kNN search, and
+a terms aggregation.
+
+[source,console]
+----
+GET example-index/_search
+{
+    "query": {
+        "term": {
+            "text": "rrf"
+        }
+    },
+    "knn": {
+        "field": "vector",
+        "query_vector": [3],
+        "k": 5,
+        "num_candidates": 5
+    },
+    "rank": {
+        "rrf": {
+            "window_size": 5,
+            "rank_constant": 1
+        }
+    },
+    "size": 3,
+    "aggs": {
+        "int_count": {
+            "terms": {
+                "field": "integer"
+            }
+        }
+    }
+}
+----
+// TEST[continued]
+
+And we receive the response with ranked `hits` and the
+terms aggregation result. Note that `_score` is replaced by
```

Comment:
Note that the _score is null, while _rank has a value which is the output of RRF?